app/src/controllers/remoteHost.js

```diff
@@ -0,0 +1,34 @@
+import { generateHash, compareWithHash } from '../utils/bcrypt.js';
+import { dbAddAPIKey, dbGetAPIKeyList } from '../db/queries.js';
+
+const addAPIKey = async (req, res, next) => {
+    try {
+        const { apiKey, name } = req.body;
+        const hash = await generateHash(apiKey);
+        const apiKeyData = {hash};
+
+        if (name) {
+            apiKeyData.name = name;
+        }
+
+        await dbAddAPIKey(apiKeyData);
+        res.status(201).send();
+    } catch(error) {
+        next(error);
+    }
+};
```

Comment:
Shouldn't this route generate the API key itself rather than letting someone pass in their own? Then we can pass back the newly generated key and display it to the user one time. After that it is only stored as a hash and not even we have access to it.

Reply:
No, it wasn't the user would pass in their own API key. I meant to modify the route to call the generation method, and then pass it back to the frontend – I simply forgot that part.

Reply:
Okay, added and we return the API key – this method was changed to a `GET` since we're not passing any data. The name can be changed in another method. See next comment (not in conversation).
![Screenshot 2023-11-04 at 3 48 44 PM](https://github.com/Sundial-Inc/server/assets/10123446/24be38f0-72f4-4fdc-81e5-f109017bd7f0)
![Screenshot 2023-11-04 at 3 50 28 PM](https://github.com/Sundial-Inc/server/assets/10123446/607ff4f6-7287-4e2e-b3b4-40537ba40d9c)

Reply:
Sigh. Not thinking straight. Generating an API key is a `POST` method and adding a name is a `PUT` method. Adding those changes.